LabExT/Measurements/InsertionLossSteppedSweep.py

```diff
@@ -0,0 +1,174 @@
+        stabilization_time = parameters.get('stabilization time').value # EM
+        averaging_time = parameters.get('averaging time').value # EM
+        laser_power = parameters.get('laser power').value
+        pm_range = parameters.get('powermeter range').value
+
+        # get instrument pointers
+        self.instr_pm = instruments['Power Meter']
+        self.instr_laser = instruments['Laser']
+
+        # open connection to Laser & PM
+        self.instr_laser.open()
+        self.instr_pm.open()
+
+        # clear errors
+        self.instr_laser.clear()
+        self.instr_pm.clear()
+
+        # Ask minimal possible wavelength
+        min_lambda = float(self.instr_laser.min_lambda)
+
+        # Ask maximal possible wavelength
+        max_lambda = float(self.instr_laser.max_lambda)
+
+        # change the minimal & maximal wavelengths if necessary
+        if start_lambda < min_lambda or start_lambda > max_lambda:
+            start_lambda = min_lambda
+            parameters['wavelength start'].value = start_lambda
+            self.logger.warning('start_lambda has been changed to smallest possible value ' + str(min_lambda))
+
+        if end_lambda > max_lambda or end_lambda < min_lambda:
+            end_lambda = max_lambda
+            parameters['wavelength stop'].value = end_lambda
+            self.logger.warning('end_lambda has been changed to greatest possible value ' + str(max_lambda))
+
+        # write the measurement parameters into the measurement settings
+        for pname, pparam in parameters.items():
+            data['measurement settings'][pname] = pparam.as_dict()
+
+        # Laser settings
+        self.instr_laser.unit = 'dBm'
+        self.instr_laser.power = laser_power
+        self.instr_laser.wavelength = start_lambda
+        number_of_points = int((end_lambda - start_lambda) / lambda_step) + 1
+
+        # PM settings
+        self.instr_pm.averagetime = averaging_time
+        self.instr_pm.wavelength = center_wavelength
+        self.instr_pm.range = pm_range
+        self.instr_pm.unit = 'dBm'
+
+        time.sleep(1) # allow laser to move to start wavelength
+
+        # inform user
+        self.logger.info(f"Sweeping over {number_of_points:d} samples "
+                         f"at {self.instr_pm.averagetime:e}s sampling period.")
+
+        power_data = list()
+        wavelengths = np.linspace(start_lambda,end_lambda,number_of_points).tolist()
+
+        self.instr_laser.enable = True # turn on laser
+
+        # STARTET DIE MOTOREN!
+        for idx, wavelength in enumerate(wavelengths):
+            self.logger.info("Taking a power measurement at {} nm.".format(wavelength))
+
+            # set laser wavelength
+            self.instr_laser.wavelength = wavelength
+            
+            # wait for the laser to stabilize and let one averaging time pass before taking a sample
+            time.sleep(stabilization_time + averaging_time)
```

Comment:
See comment above. You can possibly get rid of the stabilization time param as well.